core/events.go

```diff
@@ -49,3 +52,318 @@ type ChainSideEvent struct {
 }
 
 type ChainHeadEvent struct{ Block *types.Block }
+
+type NewTxsFeed struct {
+	sync.RWMutex
+	subs []chan<- NewTxsEvent
+}
+
+func (f *NewTxsFeed) Close() {
+	f.Lock()
+	for _, sub := range f.subs {
+		close(sub)
+	}
+	f.subs = nil
+	f.Unlock()
+}
```

Comment:
I generally prefer doing a `defer f.Unlock()` unless it becomes a performance issue. The problem is that if we `recover()` a panic while a mutex is locked then the `Unlock()` will never get called and we'll be eternally locked until the process restarts.

Also, it's cleaner in that if we add a `return` somewhere in the function in the future then we don't have to worry about adding an `Unlock()` there too.